target_compile_features specified unknown feature

### Version of Boost

develop

### Steps necessary to reproduce the problem

We just started getting this error in CI:

```
CMake Error at libs/json/CMakeLists.txt:57 (target_compile_features):
  target_compile_features specified unknown feature "_SCL_SECURE_NO_WARNINGS"
  for target "boost_json".
```

The complete output: https://github.com/boostorg/url/actions/runs/5336814814

### All relevant compiler information

The error happens in MSVC 14.34, MSVC 14.29, MinGW, and clang-cl.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -54,7 +54,7 @@ endif()
 
 function(boost_json_setup_properties target)
     if(MSVC)
-        target_compile_features(${target} PRIVATE _SCL_SECURE_NO_WARNINGS)
+        target_compile_definitions(${target} PRIVATE _SCL_SECURE_NO_WARNINGS)
     endif()
     target_compile_features(${target} PUBLIC cxx_constexpr)
     target_compile_definitions(${target} PUBLIC BOOST_JSON_NO_LIB=1)

